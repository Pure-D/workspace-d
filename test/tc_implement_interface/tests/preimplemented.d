module preimplemented;

class Element : Node, Node2
{
	void wronglyImplemented()
	{
	}

	string name() @property
	{
		return "foo";
	}
}

interface Node
{
	void addChild(Node n);
	bool wronglyImplemented();
	string name() @property;
}

interface Node2
{
	void removeChild(Node n);
}
