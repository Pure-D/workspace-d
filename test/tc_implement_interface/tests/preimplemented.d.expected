implement 42
void addChild(Node n)
bool wronglyImplemented()
!string name
void removeChild(Node n)